src/lib.rs

```diff
@@ -55,39 +42,68 @@ pub fn run(args: &clap::ArgMatches, current_dir: &PathBuf) -> Result<Vec<Output>
             None => continue,
         };
 
+        if is_fix {
+            fix_output.print_processing_info(&fe);
+        } else {
+            check_output.print_processing_info(&fe);
+        }
+
         let mut lines = get_line_entries(&fe, strs);
-        let mut backup_file = None;
 
-        // run fixers & write results to file
         let mut result = checks::run(&lines, &skip_checks);
+        // run fixers & write results to file
         if is_fix && !result.is_empty() && fixes::run(&mut result, &mut lines, &skip_checks) > 0 {
-            // create backup copy unless user specifies not to
             let should_backup = !args.is_present("no-backup");
+            // create backup copy unless user specifies not to
             if should_backup {
-                backup_file = Some(fs_utils::backup_file(&fe)?.into_os_string());
+                let backup_file = fs_utils::backup_file(&fe)?.into_os_string();
+                fix_output.print_backup(&backup_file);
             }
 
             // write corrected file
             fs_utils::write_file(&fe.path, lines)?;
         }
 
-        outputs.push(Output::new(fe, backup_file, result, mode));
+        // This shouldn't be printed to Fox when combined with quiet mode
+        if !(is_fix && is_quiet_mode) {
+            let is_last_file = i == file_paths.len() - 1;
```

Comment:
Can we move this code to the `check_output.print_warnings` method?

Reply:
> Can we move this code to the `check_output.print_warnings` method?

We could make the same method in `fix_output`, but it is not entirely correct to add it inside the `check_output` module.
Thus, I just decided to reduce the amount of duplicate code, perhaps it is better to duplicate than to complicate the logic and it will be better 🤔

Reply:
Ie we can make something like this:
```
        if is_fix {
            fix_output.print_warnings(&result, is_last_file);
        } else {
            check_output.print_warnings(&result, is_last_file)
        }
```

or passing a parameter to the enabled commit, which will confuse the logic a bit (but will reduce conditions):
```
     check_output.print_warnings(&result, fix_enabled, is_last_file);
```

@mgrachev what do you think? 🤔

Reply:
I meant this line:
```rust
let is_last_file = i == file_paths.len() - 1;
```
Can we define the last line in the `check_output.print_warnings` method?
If not, let's leave it as it was 🙂

Reply:
> I meant this line:
> 
> ```rust
> let is_last_file = i == file_paths.len() - 1;
> ```
> 
> Can we define the last line in the `check_output.print_warnings` method?

Oh, sorry for misunderstanding 🙂 
I made a commit, please take a look at implementation option for this

Reply:
Excellent implementation 👍